api/proto/teleport/legacy/types/types.proto

```diff
@@ -5955,6 +5957,30 @@ message PluginDiscordSettings {
   map<string, DiscordChannels> role_to_recipients = 1;
 }
 
+// PluginEntraIDSettings defines settings for the Entra ID sync plugin
+message PluginEntraIDSettings {
+  option (gogoproto.equal) = true;
```

Comment:
Do we want to use it as a plugin or do we want to use it as an integration similarly to AWS?
I think the second is preferred as it's similar to aws

Reply:
> Do we want to use it as a plugin or do we want to use it as an integration similarly to AWS?
> I think the second is preferred as it's similar to aws

That's a good question. I looked into it, and I'm not fully sure what the role of an `Integration` is. `Plugin` came first, then `Integration` was created at some point.  There is only a single `Integration`, that is AWS OIDC, but many `Plugin`s. Integrations like Okta and Jamf run as either:
* An independent service
* A `Plugin`.

and I would argue they are closer to what we're implementing for Entra for now, as we're importing RBAC resources and not compute resources that the AWS integration mostly focuses on.

One benefit is that hosted plugins are now enabled for any enterprise deployment and the use of them does not require to spin up a discovery service instance.

Reply:
> Don't use gogo, use derive to generate the equal funcs

I'm afraid all `Plugin` types are using `gogoproto.equal` currently, do we want to deviate from that at this point?

Reply:
I was thinking about the same, but then Okta is also defined as a plugin 🤷‍♂️ 

@r0mant ?

Reply:
@justinas plugins are expected to run on auth server and do some work.
Integrations are expected to be used by different components like the discovery_service to be able to access to the Azure/AWS APIs directly via an OIDC token signed by Teleport.

I think we should move to an integration tho

Reply:
I'm still failing to fully grasp the distinction, but I'll defer my judgement (let's see if Roman has any input).

Integration + Discovery Config workflow seems to have some rough UX edges, see https://github.com/gravitational/access-graph/issues/731 , though they are probably solvable.

Reply:
Had some time to think about this.

Entra *being run* as either a Plugin or a standalone host follows the established pattern for Jamf and Okta, as well as the upcoming GitLab TAG integration.

If we'd like to utilize the same "integration" credentials to discover Azure resources (which is out of scope for now), we only need Auth server to be able to utilize its information for signing the OIDC tokens - it should be able to do that whether the info is in an `Integration` or a `Plugin`.

IMO having this as a plugin is the most straightforward way. Let me know if I'm missing any hard blockers for this approach. @tigrato @jakule

Reply:
No preferences as long as it works really. @tigrato ?

Reply:
Can't we simply use an integration to provide access to Entra/Azure OIDC token - it's what an integration does as it doesn't run any code -  while the sync code runs as a plugin. This would simplify future changes with supporting Azure discovery in the cloud as we don't need to maintain 2 pieces of code for the same thing.

My suggestion is:
- add an integration: provides dynamic credentials signed by teleport to be used for azure/entra access
- add a plugin: leverages the integration to generate the OIDC credentials it needs to access Entra and polls access lists from it and syncs to Teleport.